### PR TITLE
boot: split MakeBootable tests into their own file

### PR DESCRIPTION
**Pull request by anonymouse64**
This is stacked on top of https://github.com/snapcore/snapd/pull/7973 and the only relevant commit is: https://github.com/snapcore/snapd/commit/e13374bd21a5ea55d738fb2f9e9c418d34f5923a.

Easier to do this now than later and makes the test files smaller. (I have a nagging feeling that the boot package is going to become the new snapstate, and thus it's worthwhile to prevent another snapstate_test.go from coming into existence :smile: )

### CODE REVIEW

File: boot/makebootable_test.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type makebootableSuite struct {
```
Comment (pedronis) [inline boot/makebootable_test.go]: nitpick: makeBootableSuite

Comment (anonymouse64) [reply boot/makebootable_test.go]: done

File: boot/makebootable_test.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type makeBootableSuite struct {
+	bootSetSuite
```
Comment (mvo5) [inline boot/makebootable_test.go]: I think we need `baseBootSetSuite` here or all the tests from bootSetSuite will run twice.

Comment (pedronis) [reply boot/makebootable_test.go]: yes, I missed this

Comment (bboozzoo) [reply boot/makebootable_test.go]: Pushed a fix

File: boot/boot_test.go [hunk 3]
```diff
@@ -54,24 +55,22 @@ func (s *baseBootSetSuite) SetUpTest(c *C) {
 	s.AddCleanup(restore)
 
 	s.bootdir = filepath.Join(dirs.GlobalRootDir, "boot")
+
+	s.bootloader = bootloadertest.Mock("mock", c.MkDir())
+	bootloader.Force(s.bootloader)
```
Comment (pedronis) [inline boot/boot_test.go]: maybe we should turn this two lines into a mockBootloader method and call it from the two suite that care, then we don't need the extra Force(nil) and also setup is done the same way in all suites

**Pull request merged**